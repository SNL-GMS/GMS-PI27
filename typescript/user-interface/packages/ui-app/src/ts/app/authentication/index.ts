import { ianAuthenticator } from './authentication';
import type { Authenticator } from './types';

export const authenticator: Authenticator = ianAuthenticator;
