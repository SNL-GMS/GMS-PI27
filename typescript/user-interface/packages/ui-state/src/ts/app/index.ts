export * from './api';
export * from './history';
export * from './hooks';
export * from './loader';
export * from './processing';
export * from './query';
export * from './redux-provider';
export * from './sanitization';
export * from './state';
export * from './store';
export * from './subscription';
export * from './util';
