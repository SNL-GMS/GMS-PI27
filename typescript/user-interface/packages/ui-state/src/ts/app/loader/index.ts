export * from './actions';
export * from './loader-reducer';
