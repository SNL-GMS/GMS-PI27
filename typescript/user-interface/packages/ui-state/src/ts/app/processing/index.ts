export * from './beamforming';
export * from './input-channels-by-prioritization';
export * from './validate-channels';
