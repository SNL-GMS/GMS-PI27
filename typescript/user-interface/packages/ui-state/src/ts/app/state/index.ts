export * from './actions';
export * from './analyst';
export * from './common';
export * from './events';
export * from './fks';
export * from './keypress';
export * from './map';
export * from './operations';
export * from './signal-detections';
export * from './station-properties-configuration';
export * from './user-session';
export * from './waveform';
export * from './workflow';
