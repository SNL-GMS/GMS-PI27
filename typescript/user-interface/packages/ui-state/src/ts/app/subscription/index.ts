import * as Subscription from './subscription';

export * from './types';
export { Subscription };
