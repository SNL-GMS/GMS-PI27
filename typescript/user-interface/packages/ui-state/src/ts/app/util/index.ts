export * from './beamforming-util';
export * as ChannelFactory from './channel-factory';
export * from './channel-segment-util';
export * from './channel-util';
export * from './data-segment-util';
export * from './react-reducer-util';
export * from './ui-rotation-processor';
export * from './ui-waveform-masking-util';
