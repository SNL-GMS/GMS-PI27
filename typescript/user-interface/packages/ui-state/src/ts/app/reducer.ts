import { combineReducers } from '@reduxjs/toolkit';
import reduceReducers from 'reduce-reducers';

import {
  eventManagerApiSlice,
  processingConfigurationApiSlice,
  processingStationApiSlice,
  signalEnhancementConfigurationApiSlice,
  stationDefinitionSlice,
  systemEventGatewayApiSlice,
  userManagerApiSlice,
  workflowApiSlice
} from './api';
import { dataSlice } from './api/data/data-slice';
import { undoRedoReducer } from './history';
import { historySlice } from './history/history-slice';
import { loaderReducer } from './loader/loader-reducer';
import { reducer as appReducer } from './state/reducer';

const reducers = combineReducers({
  // application api (queries)
  [systemEventGatewayApiSlice.reducerPath]: systemEventGatewayApiSlice.reducer,
  [eventManagerApiSlice.reducerPath]: eventManagerApiSlice.reducer,
  [processingConfigurationApiSlice.reducerPath]: processingConfigurationApiSlice.reducer,
  [processingStationApiSlice.reducerPath]: processingStationApiSlice.reducer,
  [signalEnhancementConfigurationApiSlice.reducerPath]:
    signalEnhancementConfigurationApiSlice.reducer,
  [stationDefinitionSlice.reducerPath]: stationDefinitionSlice.reducer,
  [userManagerApiSlice.reducerPath]: userManagerApiSlice.reducer,
  [workflowApiSlice.reducerPath]: workflowApiSlice.reducer,
  // undo/redo stack and history
  [historySlice.name]: historySlice.reducer,
  // data state
  [dataSlice.name]: dataSlice.reducer,
  // application state
  app: appReducer
});

export const reducer: typeof reducers = reduceReducers(reducers, loaderReducer, undoRedoReducer);
