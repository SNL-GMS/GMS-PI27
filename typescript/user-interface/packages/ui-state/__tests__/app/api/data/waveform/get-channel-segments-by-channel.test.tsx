import { PD01ChannelVersionReference } from '@gms/common-model/__tests__/__data__';
import cloneDeep from 'lodash/cloneDeep';
import type { AnyAction } from 'redux';
import type { MockStoreCreator } from 'redux-mock-store';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { GetChannelSegmentsByChannelQueryArgs } from '../../../../../src/ts/app/api/data/waveform/get-channel-segments-by-channel';
import {
  addGetChannelSegmentsByChannelMatchReducers,
  getChannelSegmentsByChannel,
  getChannelSegmentsByChannelQuery
} from '../../../../../src/ts/app/api/data/waveform/get-channel-segments-by-channel';
import type { AppState } from '../../../../../src/ts/app/store';
import { getStore } from '../../../../../src/ts/app/store';
import { appState } from '../../../../test-util';

jest.mock('../../../../../src/ts/workers/api', () => {
  const actual = jest.requireActual('../../../../../src/ts/workers/api');
  return {
    ...actual,
    fetchChannelSegmentsByChannel: jest.fn(async () =>
      Promise.reject(new Error('Rejected fetchChannelSegmentsByChannel'))
    )
  };
});

const fiveMinutes = 300;
const endTimeSecs = 123456789;
const startTimeSecs = endTimeSecs - fiveMinutes;

const waveformQueryChannelInput: GetChannelSegmentsByChannelQueryArgs = {
  channels: [PD01ChannelVersionReference],
  startTime: startTimeSecs,
  endTime: endTimeSecs
};

describe('Get Channel Segments for Channels', () => {
  it('have defined', () => {
    expect(getChannelSegmentsByChannelQuery.shouldSkip).toBeDefined();
    expect(getChannelSegmentsByChannel).toBeDefined();
    expect(addGetChannelSegmentsByChannelMatchReducers).toBeDefined();
  });

  it('build a builder using addGetChannelSegmentsByChannelMatchReducers', () => {
    const builderMap: any[] = [];
    const builder: any = {
      addCase: (k, v) => {
        builderMap.push(v);
        return builder;
      },
      addMatcher: (k, v) => {
        builderMap.push(v);
        return builder;
      }
    };

    addGetChannelSegmentsByChannelMatchReducers(builder);
    expect(builderMap).toMatchSnapshot();

    // eslint-disable-next-line prefer-const
    let state = { queries: { getChannelSegmentsByChannel: {} } };
    // eslint-disable-next-line prefer-const
    let action = {
      meta: {
        requestId: getChannelSegmentsByChannelQuery.idGenerator({
          channels: [{ name: 'channelName', effectiveAt: startTimeSecs }],
          startTime: startTimeSecs,
          endTime: endTimeSecs
        }),
        arg: { channels: [{ name: 'channelName' }] }
      },
      payload: []
    };
    builderMap[0](state, action);
    expect(state).toMatchSnapshot();
    builderMap[1](state, action);
    expect(state).toMatchSnapshot();
    builderMap[2](state, action);
    expect(state).toMatchSnapshot();
  });

  it('can determine when to skip query execution', () => {
    expect(
      getChannelSegmentsByChannelQuery.shouldSkip({
        ...waveformQueryChannelInput,
        startTime: null
      })
    ).toBeTruthy();
    expect(
      getChannelSegmentsByChannelQuery.shouldSkip({
        ...waveformQueryChannelInput,
        endTime: null
      })
    ).toBeTruthy();
    expect(
      getChannelSegmentsByChannelQuery.shouldSkip({
        ...waveformQueryChannelInput,
        channels: []
      })
    ).toBeTruthy();
    expect(getChannelSegmentsByChannelQuery.shouldSkip(waveformQueryChannelInput)).toBeFalsy();
  });

  it('will not execute query if the args are invalid', async () => {
    const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

    const store = mockStoreCreator(appState);

    await store.dispatch(
      getChannelSegmentsByChannel({ ...waveformQueryChannelInput, channels: [] }) as any
    );

    // results should have empty arrays since current interval is not set
    expect(store.getActions()).toHaveLength(0);
  });

  it('will not execute query if the current interval is not defined', async () => {
    const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

    const store = mockStoreCreator(appState);

    await store.dispatch(getChannelSegmentsByChannel(waveformQueryChannelInput) as any);

    // results should have empty arrays since current interval is not set
    expect(store.getActions()[store.getActions().length - 1].type).toEqual(
      'channelSegment/getChannelSegmentsByChannel/rejected'
    );
    expect(store.getActions()[store.getActions().length - 1].payload).toMatchInlineSnapshot(
      `[Error: Rejected fetchChannelSegmentsByChannel]`
    );
  });

  it('can handle rejected state', async () => {
    const mockStoreCreator: MockStoreCreator<AppState, AnyAction> = createMockStore([thunk]);

    const realStore = getStore();

    const state = cloneDeep(realStore.getState());
    state.app.workflow.timeRange.startTimeSecs = 4;
    state.app.workflow.timeRange.endTimeSecs = 6;

    const store = mockStoreCreator(state);

    await store.dispatch(getChannelSegmentsByChannel(waveformQueryChannelInput) as any);

    expect(store.getActions()[store.getActions().length - 1].type).toEqual(
      'channelSegment/getChannelSegmentsByChannel/rejected'
    );
  });
});
