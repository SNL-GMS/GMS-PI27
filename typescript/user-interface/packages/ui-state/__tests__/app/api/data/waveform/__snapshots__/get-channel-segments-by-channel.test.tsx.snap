// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Channel Segments for Channels build a builder using addGetChannelSegmentsByChannelMatchReducers 1`] = `
[
  [Function],
  [Function],
  [Function],
]
`;

exports[`Get Channel Segments for Channels build a builder using addGetChannelSegmentsByChannelMatchReducers 2`] = `
{
  "queries": {
    "getChannelSegmentsByChannel": {
      "startTime:123456489-endTime:123456789/channels:channelName": {
        "startTime:123456489-endTime:123456789/channels:channelName": {
          "arg": {
            "channels": [
              {
                "name": "channelName",
              },
            ],
          },
          "attempts": undefined,
          "error": {},
          "status": "pending",
          "time": undefined,
        },
      },
    },
  },
}
`;

exports[`Get Channel Segments for Channels build a builder using addGetChannelSegmentsByChannelMatchReducers 3`] = `
{
  "queries": {
    "getChannelSegmentsByChannel": {
      "startTime:123456489-endTime:123456789/channels:channelName": {
        "startTime:123456489-endTime:123456789/channels:channelName": {
          "arg": {
            "channels": [
              {
                "name": "channelName",
              },
            ],
          },
          "attempts": 1,
          "error": {},
          "status": "fulfilled",
          "time": 0.25,
        },
      },
    },
  },
}
`;

exports[`Get Channel Segments for Channels build a builder using addGetChannelSegmentsByChannelMatchReducers 4`] = `
{
  "queries": {
    "getChannelSegmentsByChannel": {
      "startTime:123456489-endTime:123456789/channels:channelName": {
        "startTime:123456489-endTime:123456789/channels:channelName": {
          "arg": {
            "channels": [
              {
                "name": "channelName",
              },
            ],
          },
          "attempts": 2,
          "error": {},
          "status": "rejected",
          "time": 0.25,
        },
      },
    },
  },
}
`;
